### PR TITLE
IGNITE-17676 Fixed race in AwaitTasksCompletionExecutor that causes tests to be flaky

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-17676

### CODE REVIEW

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/AwaitTasksCompletionExecutor.java [hunk 1]
```diff
@@ -50,9 +50,9 @@ class AwaitTasksCompletionExecutor implements Executor {
     public void execute(Runnable command) {
         CompletableFuture<?> future = new CompletableFuture<>();
 
-        future.whenComplete((o, throwable) -> updateHeartbeat.run());
+        CompletableFuture<?> heartbeatFuture = future.whenComplete((o, throwable) -> updateHeartbeat.run());
 
-        pendingTaskFutures.add(future);
+        pendingTaskFutures.add(heartbeatFuture);
```
Comment (tkalkirill) [inline modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/AwaitTasksCompletionExecutor.java:55]:
```suggestion
        pendingTaskFutures.add(future.whenComplete((o, throwable) -> updateHeartbeat.run()));
```

**Review by tkalkirill — APPROVED** [review]

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/AwaitTasksCompletionExecutor.java (hunk 1 above)
Comment (ibessonov) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/AwaitTasksCompletionExecutor.java:55]: Why do you think I should inline this statement? We would have 3 methods call in a single line, that's a lot and kinda unnecessary

Comment (tkalkirill) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/AwaitTasksCompletionExecutor.java:55]: It's just an offer, you can reject it

Comment (ibessonov) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/AwaitTasksCompletionExecutor.java:55]: Ok, I just wanted to know your motivation

Comment (tkalkirill) [reply modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/AwaitTasksCompletionExecutor.java:55]: Just fewer lines of code

**Pull request merged**